app/src/layouts/DashboardLayout/DashboardContent.js

```diff
@@ -172,8 +172,13 @@ const DashboardContent = () => {
               {...activeModals.joinWorkspaceModal.props}
             />
           ) : null}
-
-          <SharingModal />
+          {activeModals.sharingModal.isActive ? (
+            <SharingModal
+              isOpen={activeModals.sharingModal.isActive}
+              toggleModal={() => dispatch(actions.toggleActiveModal({ modalName: "sharingModal" }))}
+              {...activeModals.sharingModal.props}
+            />
+          ) : null}
```

Comment:
This is a common sharing modal we'll be using. Currently we're using this for rules but later this will be used for session replays too and there are multiple touch points to toggle share modal. Instead of creating a local state for each screen/instance we can keep the sharing modal on top level